core/domain/state_domain.py

```diff
@@ -522,8 +522,8 @@ def create_default_interaction(cls, default_dest_state_name):
             cls._DEFAULT_INTERACTION_ID, {}, [], default_outcome, [], [], {})
 
 
-    def get_all_html_content_strings(self):
-        """Get all html content strings in the interaction.
+    def get_all_html_strings(self):
```

Comment:
I think this method needs to be updated -- there should be a more programmatic way to figure out which items are HTML. Could you replace all the special casing with a general computation to figure out automatically which fields are HTML?

If you want to precompute that information and store it as a class-level constant here, that's fine (and probably preferable), but add a backend test that computes it from scratch and verifies that the result matches this constant.

In the end, this function should just iterate over that list and not have a bunch of "if self.id == ..." or "if rule_type == ..." clauses.

Reply:
Done.